### PR TITLE
Remove early stopping by setting epoch number to 501, set default epo…

### PR DESCRIPTION
**Pull request by A7exSchin**
This PR removes early stopping by setting the counter to 501 (over the default max epochs)
Also the default epoch number was set to 350 to accommodate Tongs proposed max epochs.

### CODE REVIEW

**Review by JWG-1 — APPROVED** [review]

**Comment by JWG-1** [comment]
works on my machine, stops after 350 epochs (0-349 epoch)

**Review by EhrenbruderRoschtbraten — APPROVED** [review]

File: run.py [hunk 1]
```diff
@@ -412,7 +412,7 @@ def fit(self):
 				if kill_cnt % 10 == 0 and self.p.gamma > 5:
 					self.p.gamma -= 5 
 					self.logger.info('Gamma decay on saturation, updated value of gamma: {}'.format(self.p.gamma))
-				if kill_cnt > 25:
+				if kill_cnt > 501:
```
Comment (1Maxnet1) [inline run.py]: I would prefer implementing an parameter here as well, so we can easily change the number of epochs if e.g. requested by tong

**Review by JWG-1 — APPROVED** [review]

File: run.py (hunk 1 above)
Comment (A7exSchin) [reply run.py]: I added two parameters:

1. A parameter to enable early stopping if needed.
2. A parameter to set the amount of epochs for early stopping, default is 25

File: run.py [hunk 2]
```diff
@@ -454,7 +455,8 @@ def fit(self):
 	parser.add_argument('-gcn_drop',	dest='dropout', 	default=0.1,  	type=float,	help='Dropout to use in GCN Layer')
 	parser.add_argument('-hid_drop',  	dest='hid_drop', 	default=0.3,  	type=float,	help='Dropout after GCN')
 	parser.add_argument('-disable_gnn_encoder', dest='disable_gnn_encoder', default=False, type=bool, help='Disables the GNN encoder layer')
-
+	parser.add_argument('-enable_early_stopping', dest='enable_early_stopping', default=False, type=bool, help='Enables early stopping with after the amount of epochs that were set.')
```
Comment (1Maxnet1) [inline run.py]: The with in the description is too much imho
```suggestion
	parser.add_argument('-enable_early_stopping', dest='enable_early_stopping', default=False, type=bool, help='Enables early stopping after the amount of epochs that were set.')
```

File: run.py [hunk 3]
```diff
@@ -454,7 +455,8 @@ def fit(self):
...
+	parser.add_argument('-set_early_stopping_limit', dest='early_stopping_limit', default=25, type=int, help='Sets the amount of epochs after which early stopping can occurr.')
```
Comment (1Maxnet1) [inline run.py]: I would remove the set of the parameter name and occur has an r to much
```suggestion
	parser.add_argument('-early_stopping_limit', dest='early_stopping_limit', default=25, type=int, help='Sets the amount of epochs after which early stopping can occurr.')
```

**Review by 1Maxnet1 — COMMENTED** [review]
Some little comments

File: run.py [hunk 4]
```diff
@@ -416,7 +416,8 @@ def fit(self):
 				if kill_cnt % 10 == 0 and self.p.gamma > 5:
 					self.p.gamma -= 5 
 					self.logger.info('Gamma decay on saturation, updated value of gamma: {}'.format(self.p.gamma))
-				if kill_cnt > 25:
+
+				if self.p.enable_early_stopping & kill_cnt > self.p.early_stopping_limit:
```
Comment (1Maxnet1) [inline run.py]: Guess you meant "and" (the boolean operator) and not the bit-wise and (&) ?
```suggestion
				if self.p.enable_early_stopping and kill_cnt > self.p.early_stopping_limit:
```

**Review by 1Maxnet1 — COMMENTED** [review]
One last thing :)

File: run.py (hunk 4 above)
Comment (A7exSchin) [reply run.py]: Yep thanks

**Review by 1Maxnet1 — APPROVED** [review]
LGTM, Let's get it merged

**Pull request merged**